src/PrimaryGeneration.cpp

```diff
@@ -463,3 +656,51 @@ void PrimaryGeneration::MARLEYGeneratePrimaries(G4Event* event)
 }
 
 }
+
+ROOT::Math::SMatrix< double, 3 > PrimaryGeneration::Rotation_Matrix(G4ThreeVector vec1, G4ThreeVector vec2)
```

Comment:
You should check the `CLHEP/Vector/RotationMatrix.h` file and CLHEP documentation.  They are likely to already have a function that does this computation for you, so you could use a simple `G4RotationMatrix` object, rather than writing your own.